napari/napari
# Make point size isotropic

# Description
With vispy 0.12 (#5312) we introduced some changes to how point sizes are computed. Specifically, the `scale` is no longer used to determine the size.

The initial reason was #2213. For a long explanation (with all the caveats and complications), see vispy/vispy#2453 (specifically, [this comment has a summary of the behaviour](https://github.com/vispy/vispy/issues/2453#issuecomment-1437012154)).

While the current state not ideal (and we're trying to figure out over at vispy if we can solve it), this at least solves nasty problems like #2213. However, it means that a size of `10` is still `10` even if the scale of the layer is `5` (and not `50`).

This may "break" some code that relies on this transformation; in napari, one of the consequences is that the `new points layer` button on the viewer gui will now create tiny (or massive) points if the scale is different.

To test, try this:

```py
import napari
import numpy as np

v = napari.Viewer()
v.add_image(np.random.rand(100, 100), scale=[10, 10])
```

then click the "new points" button and try to annotate some points manually. On `main`, they'll be so small that they're invisible.

---

This also exposed another issue (which became the main point of this PR): while point sizes are currently anisotropic, this information is not properly used in many places (visualisation being the primary); not only it's mostly unused, but it's handwaved in a few places (we arbitrarily take the average size over each dimension to determine the visualised size, we only allow setting isotropic sizes from gui and the `current_size`, and so on).

Ultimately, we decided that we can't reasonably support anisotropic sizes, so we should do away with them. Most changes in this PR have to do with that.

## Type of change
